### PR TITLE
[B5] custom data fields

### PR DESCRIPTION
**Pull request by orangejenny**
## Product Description

This updates the custom fields pages for users, locations, and products.

Includes migration of the key-value widget.

This is a single PR migration, so there are no new diffs and no renaming references. Reviewers can largely ignore the "initial auto-migration" commits.

| before    | after |
| -------- | ------- |
| Screenshot 2024-04-29 at 2 12 48 PM  | Screenshot 2024-04-29 at 2 10 50 PM   |
| Screenshot 2024-04-29 at 2 12 22 PM | Screenshot 2024-04-29 at 2 11 27 PM     |
| Screenshot 2024-04-29 at 2 12 29 PM    | Screenshot 2024-04-29 at 2 11 40 PM   |
| Screenshot 2024-04-29 at 2 12 34 PM    | Screenshot 2024-04-29 at 2 22 16 PM    |

### Safety story
UI-only changes. Requesting QA since there's a fair amount of javascript in here.

### Automated test coverage

No

### QA Plan
https://dimagi.atlassian.net/browse/QA-6441

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by orangejenny** [comment]
Tests are failing because I haven't yet updated diffs for the hqwebapp files updated in this PR.

**Comment by biyeun** [comment]
from an at-a-glance UI perspective, looks great!

File: corehq/apps/custom_data_fields/templates/custom_data_fields/partials/fields_tab.html [hunk 1]
```diff
@@ -7,169 +7,165 @@
...
+
+      <div data-bind="visible: validationMode() === 'choice'">
+        <button data-bind="click: addChoice, attr: { 'disabled': !isEditable() }"
+                type="button" class="btn btn-outline-primary btn-sm">
+          <i class="fa fa-plus"></i> {% trans "Add Choice" %}
+        </button>
+        <ul data-bind="sortable: choices" class="list-unstyled validation-list">
+          <li data-bind="attr: {'data-order': _sortableOrder}">
+            <div class="input-group">
+              <span class="input-group">
```
Comment (biyeun) [inline corehq/apps/custom_data_fields/templates/custom_data_fields/partials/fields_tab.html:128]: curious why the `input-group` is a span? And then also wrapped by an `input-group` parent `div`? I generally avoid `span`s wrapping anything other than text...

File: corehq/apps/custom_data_fields/templates/custom_data_fields/partials/modal_delete.html [hunk 2]
```diff
@@ -20,7 +20,7 @@ <h4 class="modal-title" id="delete-confirm-modal-label">
         <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="{% trans_html_attr "Close" %}"></button>
       </div>
 
-      <div class="modal-body" style="line-height: 18px" data-bind="css: {hide: !hasModalDetail}">  {# todo B5: inline style #}
+      <div class="modal-body" data-bind="css: {hide: !hasModalDetail}">
```
Comment (biyeun) [inline corehq/apps/custom_data_fields/templates/custom_data_fields/partials/modal_delete.html:23]: niiiice 🎉

File: corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/bootstrap5/ui-element-key-val-list.js [hunk 3]
```diff
@@ -25,42 +25,49 @@ hqDefine('hqwebapp/js/ui_elements/bootstrap5/ui-element-key-val-list', [
...
         // Create new modal controller for this element
         var $enumModal = $('<div id="' + this.modal_id + '" class="modal fade hq-enum-modal" />');
         var $modalDialog = $('<div class="modal-dialog"/>');
         var $modalContent = $('<div class="modal-content" />');
 
-        $modalContent.prepend('<div class="modal-header"><a class="close" data-dismiss="modal">×</a><h4 class="modal-title">'
-            + this.modal_title + '</h4>' + this.sub_title + '</div>');
-        var $modalForm = $('<form class="form-horizontal hq-enum-editor" action="" />'),
+        $modalContent.prepend('<div class="modal-header">'
+            + '<h4 class="modal-title">'
```
Comment (biyeun) [inline corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/bootstrap5/ui-element-key-val-list.js:45]: i know this was here before, but i always get uneasy when I see html inside of javascript.... Always prefer a template reference approach...

File: corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5.json [hunk 4]
```diff
@@ -46,5 +46,8 @@
   },
   "corehq.tabs": {
     "in_progress": true
+  },
+  "custom_user_fields": {
+    "is_complete": true
```
Comment (biyeun) [inline corehq/apps/hqwebapp/utils/bootstrap/status/bootstrap3_to_5.json:51]: yay!

File: corehq/apps/custom_data_fields/templates/custom_data_fields/partials/fields_tab.html (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/custom_data_fields/templates/custom_data_fields/partials/fields_tab.html]: unintentional, thanks - 790a57b276b

**Review by biyeun — APPROVED** [review]

**Pull request merged**